client.go

```diff
@@ -306,34 +315,48 @@ func (client *Client) eventFromMessage(message string, level Level) *Event {
 }
 
 func (client *Client) eventFromException(exception error, level Level) *Event {
-	if exception == nil {
-		exception = usageError{fmt.Errorf("%s called with nil error", callerFunctionName())}
+	err := exception
+	if err == nil {
+		err = usageError{fmt.Errorf("%s called with nil error", callerFunctionName())}
 	}
 
-	stacktrace := ExtractStacktrace(exception)
+	event := &Event{Level: level}
 
-	if stacktrace == nil {
-		stacktrace = NewStacktrace()
+	for i := 0; i < maxErrorDepth && err != nil; i++ {
+		event.Exception = append(event.Exception, Exception{
+			Value:      err.Error(),
+			Type:       reflect.TypeOf(err).String(),
+			Stacktrace: ExtractStacktrace(err),
+		})
+		if previous, ok := err.(interface{ Cause() error }); ok {
+			err = previous.Cause()
+		} else {
+			err = nil
+		}
 	}
 
-	cause := exception
-	// Handle wrapped errors for github.com/pingcap/errors and github.com/pkg/errors
-	if ex, ok := exception.(interface{ Cause() error }); ok {
-		if c := ex.Cause(); c != nil {
-			cause = c
-		}
+	// Add a trace of the current stack to the most recent error in a chain if
+	// it doesn't have a stack trace yet.
+	// We only add to the most recent error to avoid duplication and because the
+	// current stack is most likely unrelated to errors deeper in the chain.
+	if event.Exception[0].Stacktrace == nil {
+		event.Exception[0].Stacktrace = NewStacktrace()
 	}
 
-	event := NewEvent()
-	event.Level = level
-	event.Exception = []Exception{{
-		Value:      cause.Error(),
-		Type:       reflect.TypeOf(cause).String(),
-		Stacktrace: stacktrace,
-	}}
+	// event.Exception should be sorted such that the most recent error is last.
+	reverse(event.Exception)
+
 	return event
 }
 
+// reverse reverses the slice a in place.
```

Comment:
```suggestion
// reverse reverses the slice in place.
```

Reply:
Not a typo, `a` is the name of the argument :)

Reply:
I have 0 to contribute.
Maybe useless questions: Is there really no built-in generic array reversal function in Go? Oh, no generic?
Also, what about renaming `a` -> exceptions` or, `e`?
And finally:

```suggestion
// reverse reverses the slice 'a' in place.
```

:trollface: